lib/api/initiateMultipartUpload.js

```diff
@@ -189,6 +190,17 @@ function initiateMultipartUpload(authInfo, request, log, callback) {
         metadataStoreParams.controllingLocationConstraint = locConstraint;
         metadataStoreParams.dataStoreName = locConstraint;
 
+        const retentionObj = checkObjectLock(destinationBucket, request.headers);
+        if (retentionObj.error) {
+            return callback(retentionObj.error);
+        }
+        if (retentionObj.retentionInfo) {
+            metadataStoreParams.retentionInfo = retentionObj.retentionInfo;
+        }
+        if (retentionObj.legalHold) {
+            metadataStoreParams.legalHold = retentionObj.legalHold;
+        }
```

Comment:
I am pretty sure you can just merge the objects here for the same results
```suggestion
        metadataStoreParams = {...metadataStoreParams, ...retentionObj}
```

Reply:
😮 didn't know that trick, thanks!

Reply:
hmmm our linter seems to not like that syntax tho

Reply:
Grrr the linter doesn't like any spread notation I guess, probably needs to be updated. Object.assign will also work but doesn't look as cool IMO.